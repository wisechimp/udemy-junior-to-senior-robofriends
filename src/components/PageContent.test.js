import { shallow } from 'enzyme'
import React from 'react'
import PageContent from './PageContent'

let wrapper

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<PageContent { ...mockProps } />)
})

it('renders PageContent', () => {
  expect(wrapper).toMatchSnapshot()
})

it('filters robots correctly', () => {
  const mockPropsFilterTest = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'John',
      email: 'john@gmail.com'
    }],
    searchField: 'john',
    isPending: false
  }
  const wrapper2 = shallow(<PageContent { ...mockPropsFilterTest } />)
  expect(wrapper.instance().filteredRobots()).toEqual([])
  expect(wrapper2.instance().filteredRobots()).toEqual([{
      id: 3,
      name: 'John',
      email: 'john@gmail.com'
    }])
})