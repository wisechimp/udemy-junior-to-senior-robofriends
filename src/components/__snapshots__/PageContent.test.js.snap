// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PageContent 1`] = `ShallowWrapper {}`;
